Makefile

```diff
@@ -209,3 +221,7 @@ test-apb-provision: apb-push-to-local-broker
 test-apb-deprovision: apb-push-to-local-broker
 	oc run apb-test --rm=true --image=$(DEV_APB_IMAGE_FULL_NAME) --restart=Never --attach=true -- deprovision -e namespace=$(_TEST_PROJECT)
 .PHONY: test-apb-deprovision
+
+export-primary-store-configmap:
+	oc create configmap primary-store-configuration --from-file=./modules/os-datagrid-online-services-configuration/src/test/resources/caching-service/services-7.2.xml --output=json
```

Comment:
We should probably use a dedicated file for this as I'm sure the content will need to diverge eventually.

Reply:
So this config map should have a single key/value pair, the key equal to services-7.2.xml and the value equal to the content of the file. I'm wondering where this key is reference in this pull request. I don't see the reference anywhere. I can only see a reference to services.xml which is the default value.

Reply:
> We should probably use a dedicated file for this as I'm sure the content will need to diverge eventually.

I totally agree. 

> So this config map should have a single key/value pair, the key equal to services-7.2.xml and the value equal to the content of the file. I'm wondering where this key is reference in this pull request. I don't see the reference anywhere. I can only see a reference to services.xml which is the default value.

The maintenance model for this is very tricky. So this script prints out the ConfigMap's value that needs to be copied and pasted into the template. Yep, into this big piece of crap. At this moment it's a manual thing to do unfortunately. I guess I could develop some more sed/awk/perl magic to handle it but I wanted to gather some feedback before digging into it.

I'll describe how it works in the next comment.

Reply:
Well. I understand this. What about this solution? ....
1) the ConfigMap definition wouldn't be part of the template (remove the definition)
2) before instantiating the template the user would have to create the template via this command:
```oc create configmap primary-store-configuration --from-file=path/to/file/services-7.2.xml```
(this command actually creates the right configmap to be used by the service
3) then they would instantiate the template and it would reference the already installed configmap
4) they would reference the right key (e.g. services-7.2..xml via the parameter -p CONFIG_FILE=configs/services-7.2.xml
I'm not totally sure it will work but it's an idea. No need to cope with sed/perl/etc

Reply:
Yes, that could work. However it would be very tricky for the users. They would need to know what is the exact ConfigMap name they need to create. Of course, you probably know better than I that users don't read the docs...

Reply:
Well, it is a single command for creating the ConfigMap. The requirement could be that the config map needs to have the same name as the service. That's it. Users will always have to specify their configuration file somehow. Why not to specify as part of config map creation?

Reply:
I think we might solve this partially by using placeholders (see @ryanemerson's idea below and my [comment](https://github.com/jboss-container-images/jboss-dataservices-image/pull/58#discussion_r160386307))